commons/src/main/java/org/infinispan/commons/util/Features.java

```diff
@@ -0,0 +1,40 @@
+package org.infinispan.commons.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.Properties;
+
+import org.infinispan.commons.logging.Log;
+import org.infinispan.commons.logging.LogFactory;
+
+public class Features {
+   private static final Log log = LogFactory.getLog(Features.class);
+   static final String FEATURES_FILE = "META-INF/infinispan-features.properties";
+   private final Properties features;
+
+   public Features(ClassLoader classLoader) {
+      features = new Properties();
+      try {
+         Enumeration<URL> featureFiles = classLoader.getResources(FEATURES_FILE);
+         while (featureFiles.hasMoreElements()) {
+            URL url = featureFiles.nextElement();
+            try (InputStream is = url.openStream()) {
+               features.load(is);
+            } catch (IOException e) {
+            }
+         }
+      } catch (IOException e) {
+         // Ignore
+      }
+   }
+
+   public Features() {
+      this(Features.class.getClassLoader());
+   }
+
+   public boolean isAvailable(String featureName) {
+      return Boolean.getBoolean(featureName) || Boolean.parseBoolean(features.getProperty(featureName, "true"));
```

Comment:
So this allows an env prop to enable a feature (in spite of whatever is found in the meta-inf files) but it does not allow for disabling a feature. Is this asymmetry intended? I find it quite confusing. I'd imagine it would be less confusing if env props would act as an override for both true and false values.

Reply:
While the sysprop should indeed "override" a feature both ways, disabling an otherwise enabled feature thrrough a property is not something I see as particularly useful.